src/pudl/helpers.py

```diff
@@ -1527,6 +1528,19 @@ def dedupe_n_flatten_list_of_lists(mega_list):
     return list({item for sublist in mega_list for item in sublist})
 
 
+def flatten(xs: list) -> Generator:
+    """Flatten an irregular (arbitrarily nested) list of lists.
```

Comment:
this is the name/format that I saw in references (see link below)... but it doesn't feel descriptive enough for our conventions. how about `flatten_irregular_list_of_lists`?

Reply:
I'm not sure... I think we would want this to work for lists and sets. I very much copied/pasted what I seemed like the best option out of the many many options I found on the interwebs. Most of them used `Iterable`. I'm okay with making this more restrictive... we could always loosen it in the future if we wanted to.

Reply:
i kept this as an iterable and updates the docs and type hints.